RepoDb.Core/RepoDb/ClassMap.cs

```diff
@@ -0,0 +1,38 @@
+using System;
+using System.Linq.Expressions;
+
+namespace RepoDb
+{ 
+    public class ClassMap<T> where T : class
+    {
+        public Type EntityType { get; }
+
+        public ClassMap()
+        {
+            EntityType = typeof(T);
+        }
+        
+        protected ClassMap<T> Table(string table)
```

Comment:
Why is it protected?

Reply:
Encapsulation is an important part of a framework as you probably know., What to expose to the "client" and what to hide, so they won't misuse the framework. So by setting the method to protected it can only be used on a derive type like `CatClassMap : ClassMap<Cat>`

Reply:
Never thought that it could be an entity based mapping. Which is more make sense then to make it protected if this the way it should. Then I would assume we should make the ClassMap<T> abstract then? Though even there is no methods to be abstracted. I can manage that after this PR.

Reply:
Ah, seems I know the reason why it is not abstract. You do not want to remove that class instantiation flexibility by targeting certain types. Ignore my comment about abstract then.